lib/directory-monitor-poll.c

```diff
@@ -85,16 +106,37 @@ _rescan_directory(DirectoryMonitorPoll *self)
 {
   GError *error = NULL;
   GDir *directory = g_dir_open(self->super.real_path, 0, &error);
+  gchar *full_path = NULL;
+  GStatBuf st;
+  CollectionComparatorEntry *entry;
+
   collection_comparator_start(self->comparator);
   if (directory)
     {
       const gchar *filename;
       while((filename = g_dir_read_name(directory)))
         {
           collection_comparator_add_value(self->comparator, filename);
+
+          full_path = build_filename(self->super.real_path, filename);
```

Comment:
This block imho in the wrong level of abstraction. The poll version of directory monitor uses the `collection_comparator` inside to decide what state a file is in. (created, deleted, ...)
The part where it request `stat` and compare should be in the `collection_comparater`, and here the `collection_comparator_add_value` should hide the change inside it.